ui/src/containers/AppsTableContainer.js

```diff
@@ -1,57 +1,47 @@
 import React from 'react';
-import { sortable, SortByDirection } from '@patternfly/react-table';
+import { SortByDirection } from '@patternfly/react-table';
 import AppsTable from '../components/AppsTable';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { getApps } from '../actions/actions';
+import { reverseAppsTableSort } from '../actions/actions-ui';
 
-class AppsTableContainer extends React.Component {
+export class AppsTableContainer extends React.Component {
   constructor (props) {
     super(props);
-    this.sortBy = {};
     this.onSort = this.onSort.bind(this);
     this.onRowClick = this.onRowClick.bind(this);
   }
-  componentDidMount () {
-    this.props.getApps();
-  }
   onRowClick (event, rowId, props) {
+    console.log('On row click called');
     this.setState({ redirect: true });
   }
-  onSort (_event, index, direction) {
+  onSort (_event, index) {
     const sortedRows = this.props.apps.sort((a, b) => (a[index] < b[index] ? -1 : a[index] > b[index] ? 1 : 0));
-    this.setState({
-      sortBy: {
-        index,
-        direction
-      },
-      apps: direction === SortByDirection.asc ? sortedRows : sortedRows.reverse()
-    });
+    const sortedApps = this.props.sortBy.direction === SortByDirection.asc ? sortedRows : sortedRows.reverse();
+    this.props.reverseAppsTableSort(this.props.sortBy.direction, index, sortedApps);
```

Comment:
while this does work, however, I think it will be better if the reverse part is removed to be part of the reducer. At the moment, if you read the `REVERSE_SORT` code in reducer, it doesn't actually do anything. People will confused why it is not doing the reverse there (at least I was). Normally speaking, in redux, all the data manipulation should happen there.

Reply:
@wei-lee do you mean line 20 should be moved out to the action. The logic that currently lives in the action reverses the SortByDirection value which is mapped to props.sortBy.direction. Should this also be done in the reducer?

Reply:
agree with @wei-lee, the `onSort()` should really just be making an action call to a reducer that looks after the data manipulation

Reply:
👍 going to fix this now.